SmartDeviceLink/SDLSystemCapabilityManager.m

```diff
@@ -0,0 +1,142 @@
+//
+//  SDLSystemCapabilityManager.m
+//  SmartDeviceLink
+//
+//  Created by Nicole on 3/26/18.
+//  Copyright © 2018 smartdevicelink. All rights reserved.
+//
+
+#import "SDLSystemCapabilityManager.h"
+
+#import "SDLConnectionManagerType.h"
+#import "SDLGenericResponse.h"
+#import "SDLGetSystemCapability.h"
+#import "SDLGetSystemCapabilityResponse.h"
+#import "SDLLogMacros.h"
+#import "SDLNotificationConstants.h"
+#import "SDLRegisterAppInterfaceResponse.h"
+#import "SDLRPCResponseNotification.h"
+#import "SDLSetDisplayLayoutResponse.h"
+#import "SDLSystemCapability.h"
+#import "SDLVideoStreamingCapability.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface SDLSystemCapabilityManager ()
+
+@property (weak, nonatomic) id<SDLConnectionManagerType> connectionManager;
+@property (copy, nonatomic, nullable) SDLUpdateCapabilityHandler systemCapabilityHandler;
+
+@property (nullable, strong, nonatomic, readwrite) SDLDisplayCapabilities *displayCapabilities;
+@property (nullable, strong, nonatomic, readwrite) SDLHMICapabilities *hmiCapabilities;
+@property (nullable, copy, nonatomic, readwrite) NSArray<SDLSoftButtonCapabilities *> *softButtonCapabilities;
+@property (nullable, copy, nonatomic, readwrite) NSArray<SDLButtonCapabilities *> *buttonCapabilities;
+@property (nullable, strong, nonatomic, readwrite) SDLPresetBankCapabilities *presetBankCapabilities;
+@property (nullable, copy, nonatomic, readwrite) NSArray<SDLHMIZoneCapabilities> *hmiZoneCapabilities;
+@property (nullable, copy, nonatomic, readwrite) NSArray<SDLSpeechCapabilities> *speechCapabilities;
+@property (nullable, copy, nonatomic, readwrite) NSArray<SDLPrerecordedSpeech> *prerecordedSpeechCapabilities;
+@property (nonatomic, readwrite) BOOL vrCapability;
+@property (nullable, copy, nonatomic, readwrite) NSArray<SDLAudioPassThruCapabilities *> *audioPassThruCapabilities;
+@property (nullable, strong, nonatomic, readwrite) SDLAudioPassThruCapabilities *pcmStreamCapability;
+@property (nullable, strong, nonatomic, readwrite) SDLNavigationCapability *navigationCapability;
+@property (nullable, strong, nonatomic, readwrite) SDLPhoneCapability *phoneCapability;
+@property (nullable, strong, nonatomic, readwrite) SDLVideoStreamingCapability *videoStreamingCapability;
+@property (nullable, strong, nonatomic, readwrite) SDLRemoteControlCapabilities *remoteControlCapability;
+
+@end
+
+@implementation SDLSystemCapabilityManager
+
+#pragma mark - Lifecycle
+
+- (instancetype)initWithConnectionManager:(id<SDLConnectionManagerType>)manager {
+    self = [super init];
+    if (!self) {
+        return nil;
+    }
+
+    _connectionManager = manager;
+    [self sdl_registerForNotifications];    
+
+    return self;
+}
+
+- (void)stop {
+    SDLLogD(@"System Capability manager stopped");
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
```

Comment:
This method should not remove the observer, or else if you disconnect and reconnect, the observers will never fire, because this is not re-inited. This should clear all the internal capability data however. Or else, all this data will continue to be available while we disconnect and will still be there on a reconnect, even if it's to a different head unit.

Reply:
Fixed